source/d3d8to9_base.cpp

```diff
@@ -62,7 +64,14 @@ HRESULT STDMETHODCALLTYPE Direct3D8::QueryInterface(REFIID riid, void **ppvObj)
 		return S_OK;
 	}
 
-	return ProxyInterface->QueryInterface(riid, ppvObj);
+	HRESULT hr = ProxyInterface->QueryInterface(ConvertREFIID(riid), ppvObj);
```

Comment:
It is really valid to query a device from the base D3D interface?

Reply:
I ran a few tests and found out that you cannot query a device from the base D3D interface.  So I removed the code and anything related from the base D3D interface.